packages/core/src/services/documentLogs/computeDocumentLogs.ts

```diff
@@ -0,0 +1,88 @@
+import { and, desc, eq, isNull, sql } from 'drizzle-orm'
+
+import { Commit, DEFAULT_PAGINATION_SIZE, ErrorableEntity } from '../../browser'
+import { database } from '../../client'
+import { calculateOffset } from '../../lib/pagination/calculateOffset'
+import { DocumentLogsRepository } from '../../repositories'
+import {
+  commits,
+  documentLogs,
+  projects,
+  runErrors,
+  workspaces,
+} from '../../schema'
+import { getCommonQueryConditions } from './computeDocumentLogsWithMetadata'
+
+export function computeDocumentLogsQuery(
+  {
+    workspaceId,
+    documentUuid,
+    draft,
+    allowAnyDraft,
+    page = '1',
+    pageSize = String(DEFAULT_PAGINATION_SIZE),
+  }: {
+    workspaceId: number
+    documentUuid?: string
+    draft?: Commit
+    allowAnyDraft?: boolean
+    page?: string
+    pageSize?: string
+  },
+  db = database,
+) {
+  const repo = new DocumentLogsRepository(workspaceId, db)
+  const offset = calculateOffset(page, pageSize)
+  const scope = repo.scope.as('document_logs')
+  const fields = scope._.selectedFields
+  return db
```

Comment:
Question:
In [other places](https://github.com/latitude-dev/latitude-llm/blob/main/packages/core/src/services/documentLogs/computeDocumentLogsWithMetadata.ts#L55-L67) we do `repo.scope.where(...)` 

But I tried here and it didn't work. You know why?

Reply:
you should be able to do it since scope is a dynamic query in the case of document logs repository

Reply:
I think you can actually do that with this repository. It's even used in some of the repository's methods themselves! Although maybe you can no longer do that after adding an `.as(…)` alias.

Reply:
yeah no if you add an alias you can't do it, but you shouldn't add an alias 😅